### PR TITLE
DROOLS-2156: DMN decision table names should be presented as titles in editor screen

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2156

The "Expression Editor" screen now contains the type of expression (identical to the Navigator).

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionContainerGrid.java [hunk 1]
```diff
@@ -113,7 +122,54 @@ public ExpressionContainerGrid(final DMNGridLayer gridLayer,
...
+            @Override
+            public DMNModelInstrumentedBase asDMNModelInstrumentedBase() {
+                return hasExpression.getExpression();
+            }
+        };
+
+        return spy;
+    }
+
+    Optional<HasName> spyHasName(Optional<HasName> hasName) {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionContainerGrid.java]: Please, could we use `final Optional<HasName> hasName`

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java [hunk 2]
```diff
@@ -167,13 +176,20 @@ public void setExpression(final String nodeUUID,
...
     }
 
     private void setReturnToDRGText(final Optional<HasName> hasName) {
         hasName.ifPresent(name -> returnToDRG.setTextContent(translationService.format(DMNEditorConstants.ExpressionEditor_ReturnToDRG,
                                                                                        name.getName().getValue())));
     }
 
+    private void setExpressionTypeText(final Optional<Expression> expression) {
+        expressionType.setTextContent(expression.isPresent() ?
+                                              expression.get().getClass().getSimpleName() :
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java]: This is not ideal for multiple word names. Headings are rendered without whitespace between words:
- LiteralExpression
- DecisionTable
- FunctionDefinition

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java]: @jomarko I know.. but it is currently consistent with the Navigator in the west dock.

I'd propose a new JIRA to add i18n to both the Navigator and this change.

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java]: (in the name of "agile" development!!)

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java]: @manstis understood, I am fine with I18N issue for this, however we need to fix the renaming issue I discovered below before merging.

**Comment by manstis** [comment]
@jomarko New commit added to include ```final```.

**Comment by jomarko** [comment]
@manstis I recompile whole kie-wb-common, even more I rebased that PR with master changes, still see the same issue, I attached the video into [jira](https://issues.jboss.org/browse/DROOLS-2156).

**Comment by manstis** [comment]
@karreiro I am unable to replicate @jomarko issue; would you be able to try please?

**Comment by manstis** [comment]
@jomarko @karreiro I tried to replicate again today however it continues to work OK for me.

**Pull request merged**